### PR TITLE
`azurerm_role_assignment`: Allow for sub aliases scopes

### PR DESCRIPTION
**Pull request by favoretti**

### CODE REVIEW

**Comment by favoretti** [comment]
1 failing is transient and isn't related to the chages.

image

File: internal/services/authorization/parse/role_assignment_test.go [hunk 1]
```diff
@@ -166,6 +175,15 @@ func TestRoleAssignmentID(t *testing.T) {
 			},
 		},
 
+		{
+			// valid at subscriptions aliases scope
+			Input: "/providers/Microsoft.Subscription/aliases/my-awesome-sub/providers/Microsoft.Authorization/roleAssignments/23456781-2349-8764-5631-234567890121",
+			Expected: &RoleAssignmentId{
+				IsSubscriptionAliasLevel: true,
+				Name:                     "23456781-2349-8764-5631-234567890121",
```
Comment (manicminer) [inline internal/services/authorization/parse/role_assignment_test.go:183]: It looks like `IsSubscriptionAliasLevel` is not being tested below?

Also I think `SubscriptionAlias` should be added/tested here?

```suggestion
				Name:                     "23456781-2349-8764-5631-234567890121",
				SubscriptionAlias:        "my-awesome-sub",
```

**Review by manicminer — CHANGES_REQUESTED** [review]
Hi @favoretti, thanks for looking at this. I think overall this looks good, I'm just concerned about unit test coverage (see comment below).

**Comment by favoretti** [comment]
> Hi @favoretti, thanks for looking at this. I think overall this looks good, I'm just concerned about unit test coverage (see comment below).

@manicminer Good catch on testing `SubscriptionAlias` value, thanks for that. Not sure I follow though about `IsSubscriptionAliasLevel` not being tested below?

**Comment by manicminer** [comment]
@favoretti Towards the end of the `TestRoleAssignmentID()` func, there are checks to ensure the returned ID is populated correctly - we'll want to add checks for the `IsSubscriptionAliasLevel and `SubscriptionAlias` fields, else the test will just pass whatever their value :)

**Comment by favoretti** [comment]
@manicminer gotcha, pushed a fix.

**Review by katbyte — APPROVED** [review]
LGTM 💡

**Comment by PreciousEdmund** [comment]
Hi, I have a customer that has issues with this version though
When using this version he gets this error **_"Error: reading Subscription Alias “Hidden ID”: subscriptions.SubscriptionsClient#AliasGet: Failure responding to request: StatusCode=401 – Original Error: autorest/azure: Service returned an error. Status=401 Code=”UserNotAuthorized” Message=”User does not have access Microsoft.Subscription/aliases/write over scope providers/Microsoft.Subscription/aliases/HiddenID"_**

But when using the version 3.49.0 everything works out fine.

**Hidden ID**: To Prevent personal information

**Comment by sampathkovuri** [comment]
It seems that I am experiencing a similar error as mentioned above, but only with manually created subscriptions. I have several subscriptions that were created programmatically with Terraform, where I am able to upgrade the azurerm provider without any issues. If anyone has a solution for this problem, please inform me.

**Comment by favoretti** [comment]
So initial release code in 3.49 worked? Any ideas which release broke it?

**Comment by sampathkovuri** [comment]
> So initial release code in 3.49 worked? Any ideas which release broke it?

It works without any issues with 3.49.0, breaks with all versions above it

**Comment by PreciousEdmund** [comment]
Oh, glad i'm not the only @sampathkovuri 

Please @favoretti , can you look into this 


image

**Pull request merged**